src/libraries/System.IO.Compression/src/System/IO/Compression/ZipBlocks.cs

```diff
@@ -534,6 +552,24 @@ public static bool TryValidateBlock(BinaryReader reader, ZipArchiveEntry entry)
 
             return true;
         }
+
+        internal static ulong ConvertToUlong(uint loverAddressValue, uint higherAddressValue)
+        {
+            return checked((ulong)loverAddressValue + (((ulong)higherAddressValue) << 32));
```

Comment:
Is `checked` really necessary here? I don't think anything here can overflow.

Reply:
right